allennlp/models/model.py

```diff
@@ -116,12 +117,37 @@ def forward_on_instance(self, instance: Instance, cuda_device: int) -> Dict[str,
         outputs = self.decode(self.forward(**model_input))
 
         for name, output in list(outputs.items()):
+            # We are predicting on a single instance and we added a batch
+            # dimension, so here we remove it.
             output = output[0]
             if isinstance(output, torch.autograd.Variable):
                 output = output.data.cpu().numpy()
             outputs[name] = output
         return outputs
 
+    def forward_on_batch(self, instances: List[Instance],
+                         cuda_device: int) -> Dict[str, numpy.ndarray]:
+        """
+        Takes a list of  :class:`~allennlp.data.instance.Instance`s, converts that text into
+        arrays using this model's :class:`Vocabulary`, passes those arrays through
+        :func:`self.forward()` and :func:`self.decode()` (which by default does nothing)
+        and returns the result.  Before returning the result, we convert any
+        ``torch.autograd.Variables`` or ``torch.Tensors`` into numpy arrays.
```

Comment:
You could note in here the similarity between this method and `forward_on_instance`, that this will actually perform batched computation on the GPU.